### PR TITLE
Feature/result refactoring

### PR DESCRIPTION
**Pull request by siy**
- Cleanup and refactor error handling/codes for client API
- Added support for "rri" along with "tokenIdentifier" (this one is deprecated and soon will be removed)
- Full implementation of simple synchronous Java client for JSON RPC API

### CODE REVIEW

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java [hunk 1]
```diff
@@ -23,58 +23,71 @@
...
-		}
+			@Override
+			public boolean equals(Object obj) {
+				if (obj == this) {
+					return true;
+				}
 
-		return (obj instanceof Failure) && Objects.equals(((Failure) obj).message(), message);
-	}
+				return (obj instanceof Failure) && Objects.equals(((Failure) obj).message(), message);
```
Comment (talekhinezh) [inline radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java:84]: This should check class is the same as well as `code`. Also, weird to have anonymous class have equals, better to just create separate class.

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java [hunk 2]
```diff
@@ -23,58 +23,71 @@
...
+				return code;
+			}
 
-	@Override
-	public int hashCode() {
-		return Objects.hash(message);
-	}
+			@Override
+			public int hashCode() {
+				return Objects.hash(message);
```
Comment (talekhinezh) [inline radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java:75]: Missing `code` which would have been caught by EqualsVerifier.

**Review by talekhinezh — APPROVED** [review]

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java (hunk 1 above)
Comment (siy) [reply radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java:84]: Moved into separate class.

File: radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java (hunk 2 above)
Comment (siy) [reply radixdlt-java-common/src/main/java/com/radixdlt/utils/functional/Failure.java:75]: Fixed

**Review by LukasGasior1 — APPROVED** [review]

**Pull request merged**